numba/core/ssa.py

```diff
@@ -60,19 +62,19 @@ def _run_ssa(blocks):
         _logger.debug("Replaced assignments: %s", pformat(defmap))
         # Fix up the RHS
         # Re-associate the variable uses with the reaching definition
-        blocks = _fix_ssa_vars(blocks, varname, defmap)
+        blocks = _fix_ssa_vars(blocks, varname, defmap, cfg, df_plus)
```

Comment:
Is it worth asserting that the CFG hasn't changed on exit? CFGs are comparable with `__eq__`

Reply:
That would require recomputing the cfg and would be too expensive given that SSA is already too costly in our compilation pipeline.  We have to rely on making sure the SSA algorithm are not mutating the CFG.

Reply:
out of band, cost of a single CFG recomputation on loop exit was considered worth it